tensorflow/python/training/learning_rate_decay.py

```diff
@@ -77,10 +78,173 @@ def exponential_decay(learning_rate, global_step, decay_steps, decay_rate,
+      'ExponentialTimeDecay'
+
+  Returns:
+    A scalar `Tensor` of the same type as `learning_rate`.  The decayed
+    learning rate.
+  """
+  with ops.op_scope([initial_lr, epoch, decay_rate],
+                   name, "ExponentialTimeDecay") as name:
+    initial_lr = ops.convert_to_tensor(initial_lr, name="learning_rate")
+    decay_rate = cast(decay_rate, initial_lr.dtype)
+    exponent = exp(mul(neg(decay_rate), epoch))
+    return mul(initial_lr, exponent, name=name)
+
+def inverse_time_decay(initial_lr, epoch, decay_rate, name=None):
+  """Applies inverse time decay to the learning rate.
+
+  When training a model, it is often recommended to lower the learning rate as
+  the training progresses.  This function applies an inverse decay function
+  to a provided initial learning rate.  It requires an `epoch` value to
+  compute the decayed learning rate.  You can just pass a TensorFlow variable
+  that you increment at each training step.
+
+  The function returns the decayed learning rate.  It is computed as:
+
+  ```python
+  decayed_learning_rate = learning_rate / (1 + decay_rate * t)
+  ```
+
+  Example: decay 1/t with a rate of 0.5:
+
+  ```python
+  ...
+  epoch = tf.Variable(0, trainable=False)
+  initial_lr = 0.1
+  decay_rate = 0.5
+  learning_rate = tf.train.inverse_time_decay(initial_lr, epoch, k)
+
+  # Passing epoch to minimize() will increment it at each step.
+  learning_step = (
+      tf.GradientDescentOptimizer(learning_rate)
+      .minimize(...my loss..., global_step=epoch)
+  )
+  ```
+
+  Args:
+    initial_lr: A scalar `float32` or `float64` `Tensor` or a
+      Python number.  The initial learning rate.
+    epoch: A Python number.
+      Global step to use for the decay computation.  Must not be negative.
+    decay_rate: A Python number.  The decay rate.
+    name: String.  Optional name of the operation.  Defaults to
+      'ExponentialTimeDecay'
+
+  Returns:
+    A scalar `Tensor` of the same type as `learning_rate`.  The decayed
+    learning rate.
+  """
+
+  with ops.op_scope([initial_lr, epoch, decay_rate],
+                   name, "InverseTimeDecay") as name:
+    initial_lr = ops.convert_to_tensor(initial_lr, name="learning_rate")
+    decay_rate = cast(decay_rate, initial_lr.dtype)
+    const = cast(constant(1), initial_lr.dtype)
+    denom = add(const, mul(decay_rate, epoch))
+    return div(initial_lr, denom, name=name)
+
+def step_time_decay(learning_rate, epoch, num_steps_for_decay, decay_rate=0.5,
+               name=None):
+  """Applies step decay to the learning rate.
+
+  When training a model, it is often recommended to lower the learning rate as
+  the training progresses.  This function applies a step decay function
+  to a provided initial learning rate.  It requires a `global_step` value to
+  compute the decayed learning rate.  You can just pass a python number
+  that you increment at each training step.
+
+  The function returns the decayed learning rate.  It is computed as:
```

Comment:
Can you describe in more detail what type of behavior you want this to have?   The function below suggests that the learning rate only changes on exactly one step every epoch, which I don't think is what you wanted.